frontend/src/pages/ShootItemEditor.vue

```diff
@@ -0,0 +1,485 @@
+          if (element && element.remove) {
+            element.remove()
+          }
+        }
+        this.$instance = undefined
+      },
+      setOption (key, value) {
+        if (this.$instance) {
+          this.$instance.setOption(key, value)
+        }
+      },
+      getOption (key) {
+        if (this.$instance) {
+          this.$instance.getOption(key)
+        }
+      },
+      clearHistory () {
+        if (this.$instance) {
+          this.$instance.doc.clearHistory()
+          this.generation = this.$instance.doc.changeGeneration()
+          this.clean = true
+          this.untouched = true
+          this.conflictPath = null
+          this.historySize.undo = 0
+          this.historySize.redo = 0
+        }
+      },
+      getContent () {
+        if (this.$instance) {
+          return this.$instance.doc.getValue()
+        }
+        return ''
+      },
+      setContent (value) {
+        if (this.$instance) {
+          const editor = this.$instance
+          const doc = editor.doc
+          const cursor = doc.getCursor()
+          const {left, top} = editor.getScrollInfo() || {}
+          doc.setValue(value)
+          editor.scrollTo(left, top)
+          editor.focus()
+          if (cursor) {
+            doc.setCursor(cursor)
+          }
+          this.clearHistory()
+        }
+      },
+      update (value = this.value) {
+        if (value) {
+          this.setContent(safeDump(value))
+        }
+      },
+      resolveAction (value) {
+        this.dialog = false
+        if (isFunction(this.action.resolve)) {
+          const resolve = this.action.resolve
+          this.action.resolve = undefined
+          resolve(value)
+        }
+      },
+      confirmNavigation () {
+        this.dialog = true
+        return new Promise(resolve => {
+          assign(this.action, {
+            id: 'navigation',
+            title: 'Confirm Navigation',
+            text: 'Your changes have not been saved.<br/>Are you sure you want to leave this page?',
+            yesButtonText: 'Leave',
+            noButtonText: 'Stay',
+            resolve
+          })
+        })
+      },
+      confirmOverwrite () {
+        this.dialog = true
+        return new Promise(resolve => {
+          assign(this.action, {
+            id: 'save',
+            title: 'Confirm Overwrite',
+            text: 'Meanwhile another user or process has changed the cluster resource.<br/>Are you sure you want to overwrite it?',
+            yesButtonText: 'Save',
+            noButtonText: 'Cancel',
+            resolve
+          })
+        })
+      }
+    },
+    mounted () {
+      const value = this.$localStorage.getItem('showShootEditorWarning')
+      this.modificationWarning = value === null || value === 'true'
+      this.createInstance(this.$refs.container)
+      this.update(this.value)
+      this.refresh()
+      // clipboard
+      const vm = this
+      const clipboard = new Clipboard(vm.$refs.btnCopy.$el, {
```

Comment:
maybe use target

Reply:
target determines from which element the content to be copied is taken. We provide the content in the text callback. The button still has the focus after clicking. I now focus the editor after all toolbar buttons clicks.

Reply:
:heavy_check_mark: